devdoc/buffer_requirements.md

```diff
@@ -39,7 +39,7 @@ extern BUFFER_HANDLE BUFFER_create(const unsigned char* source, size_t size);
 
 BUFFER_create creates a new buffer from the memory at source, having size "size".
 **SRS_BUFFER_02_001: [**If source is NULL then BUFFER_create shall return NULL.**]** 
-**SRS_BUFFER_02_002: [**Otherwise, BUFFER_create shall allocate memory to hold size bytes and shall copy from source size bytes into the newly allocated memory.**]** 
+**SRS_BUFFER_02_002: [**Otherwise, BUFFER_create shall allocate memory to hold size bytes and shall copy from source size bytes into the newly allocated memory. If size parameter is 0 then buffer field of BUFFER_TAG shall still hold a valid pointer whose content is '\0' and size field shall be 0.**]** 
```

Comment:
I understand. Thinking about it again, yes, it is better to allocate 1 byte, but there is no need to set the memory to '\0', as noone should ever use that.
The other point about having this as a separate req id stands.

---
In reply to: [75432507](https://github.com/Azure/azure-c-shared-utility/pull/20#discussion_r75432507) [](ancestors = 75432507)